python/cugraph-pyg/cugraph_pyg/data/cugraph_store.py

```diff
@@ -472,42 +482,51 @@ def _get_edge_index(self, attr: CuGraphEdgeAttr) -> Tuple[TensorType, TensorType
 
-        if isinstance(attr.edge_type, str):
-            edge_type = attr.edge_type
+        if self._is_delayed:
+            src_col_name = self.__graph.renumber_map.renumbered_src_col_name
+            dst_col_name = self.__graph.renumber_map.renumbered_dst_col_name
         else:
-            edge_type = attr.edge_type[1]
+            src_col_name = self.__graph.srcCol
+            dst_col_name = self.__graph.dstCol
 
         # If there is only one edge type (homogeneous graph) then
         # bypass the edge filters for a significant speed improvement.
-        if len(self.__graph.edge_types) == 1:
-            if list(self.__graph.edge_types)[0] != edge_type:
+        if len(self.__edge_types_to_attrs) == 1:
+            if attr.edge_type not in self.__edge_types_to_attrs:
                 raise ValueError(
-                    f"Requested edge type {edge_type}" "is not present in graph."
+                    f"Requested edge type {attr.edge_type}" "is not present in graph."
                 )
 
-            df = self.__graph.get_edge_data(
-                edge_ids=None,
-                types=None,
-                columns=[self.__graph.src_col_name, self.__graph.dst_col_name],
-            )
+            df = self.__graph.edgelist.edgelist_df[[src_col_name, dst_col_name]]
+            src_offset = 0
+            dst_offset = 0
         else:
-            if isinstance(attr.edge_type, str):
-                edge_type = attr.edge_type
-            else:
-                edge_type = attr.edge_type[1]
-
-            # FIXME unrestricted edge type names
-            df = self.__graph.get_edge_data(
-                edge_ids=None,
-                types=[edge_type],
-                columns=[self.__graph.src_col_name, self.__graph.dst_col_name],
+            src_type, _, dst_type = attr.edge_type
+            src_offset = int(
+                self.__vertex_type_offsets["start"][
+                    np.searchsorted(self.__vertex_type_offsets["type"], src_type)
+                ]
+            )
+            dst_offset = int(
+                self.__vertex_type_offsets["start"][
+                    np.searchsorted(self.__vertex_type_offsets["type"], dst_type)
+                ]
+            )
+            coli = np.searchsorted(
+                self.__edge_type_offsets["type"], "__".join(attr.edge_type)
             )
 
+            df = self.__graph.edgelist.edgelist_df[
+                [src_col_name, dst_col_name, self.__graph.edgeTypeCol]
+            ]
+            df = df[df[self.__graph.edgeTypeCol] == coli]
+            df = df[[src_col_name, dst_col_name]]
+
         if self._is_delayed:
             df = df.compute()
 
-        src = self.from_dlpack(df[self.__graph.src_col_name].to_dlpack())
-        dst = self.from_dlpack(df[self.__graph.dst_col_name].to_dlpack())
+        src = self.from_dlpack(df[src_col_name].to_dlpack()) - src_offset
+        dst = self.from_dlpack(df[dst_col_name].to_dlpack()) - dst_offset
```

Comment:
Same suggestion here of moving away from dlpack to enable using managed memory.

Reply:
This was fixed in the other branch, which I just merged into this one.